src/components/Fetcher/index.js

```diff
@@ -0,0 +1,133 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Spinner from 'react-spinkit'
+import { connect } from 'react-redux'
+
+import LANGUAGE_FETCHER from 'endpoints/language'
+import PAGE_ENDPOINT from 'endpoints/page'
+import DISCLAIMER_FETCHER from 'endpoints/disclaimer'
+import LOCATION_FETCHER from 'endpoints/location'
+
+import Error from 'components/Error'
+
+import style from './style.css'
+
+function createFetcher (endpoint,
+                        createUrlOptions, createUrlTransformOptions,
+                        mapStateToProps = () => { return {} },
+                        shouldUpdate = () => false) {
+  let stateName = endpoint.name
+  let payloadName = stateName + 'Payload'
+
+  function mapStateToFetcherProps (state) {
+    let props = mapStateToProps(state)
+
+    props[payloadName] = state[stateName]
+    return props
+  }
+
+  let Fetcher = class extends React.PureComponent {
+    static propTypes = {
+      hideError: PropTypes.bool,
+      hideSpinner: PropTypes.bool
+    }
```

Comment:
maybe we also want an optional style property, so that the components can style their loading and error messages independently. If no style property is provided, take default.